controller/internal/routes/health_check_router.go

```diff
@@ -0,0 +1,145 @@
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package routes
+
+import (
+	"context"
+	health "github.com/AppsFlyer/go-sundheit"
+	"github.com/AppsFlyer/go-sundheit/checks"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/openziti/edge/controller/apierror"
+	"github.com/openziti/edge/controller/env"
+	"github.com/openziti/edge/controller/internal/permissions"
+	"github.com/openziti/edge/controller/response"
+	"github.com/openziti/edge/rest_management_api_server/operations/informational"
+	"github.com/openziti/foundation/metrics"
+	"github.com/openziti/foundation/util/concurrenz"
+	"github.com/openziti/foundation/util/errorz"
+	"github.com/openziti/sdk-golang/ziti/edge/impl"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"go.etcd.io/bbolt"
+	"time"
+)
+
+func init() {
+	r := NewHealthCheckRouter()
+	env.AddRouter(r)
+}
+
+type HealthCheckRouter struct {
+	BasePath      string
+	healthChecker health.Health
+}
+
+func NewHealthCheckRouter() *HealthCheckRouter {
+	return &HealthCheckRouter{
+		BasePath:      "/health-check",
+		healthChecker: health.New(),
+	}
+}
+
+func (r *HealthCheckRouter) Register(ae *env.AppEnv) {
+	ae.ManagementApi.InformationalHealthCheckHandler = informational.HealthCheckHandlerFunc(func(params informational.HealthCheckParams) middleware.Responder {
+		return ae.IsAllowed(r.HealthCheck, params.HTTPRequest, "", "", permissions.Always())
+	})
+
+	check, err := checks.NewPingCheck("bolt.read", &boltPinger{
+		appEnv:      ae,
+		openReadTxs: ae.GetMetricsRegistry().Gauge("bolt.open_read_txs"),
+	})
+
+	if err != nil {
+		logrus.WithError(err).Fatal("unable to create bolt.read ping check")
+		return
+	}
+
+	// TODO: once we have fabric xweb based API, move this to fabric and make check configurable
+	err = r.healthChecker.RegisterCheck(check, health.ExecutionPeriod(30*time.Second), health.InitiallyPassing(true), health.ExecutionTimeout(15*time.Second))
+	if err != nil {
+		logrus.WithError(err).Fatal("unable to register bolt.read ping check")
+		return
+	}
+}
+
+func (r *HealthCheckRouter) HealthCheck(ae *env.AppEnv, rc *response.RequestContext) {
+	results, healthy := r.healthChecker.Results()
+	if healthy {
+		rc.RespondWithEmptyOk()
+		return
+	}
+
+	var errs []error
+
+	for id, result := range results {
+		if !result.IsHealthy() {
+			if result.Error != nil {
+				errs = append(errs, result.Error)
+			} else {
+				errs = append(errs, errors.Errorf("check %v reports unhealthy state", id))
+			}
+		}
+	}
+
+	var err *errorz.ApiError
+	if len(errs) == 1 {
+		err = apierror.NewControllerUnhealthyError(errs[0])
+	} else {
+		err = apierror.NewControllerUnhealthyError(impl.MultipleErrors(errs))
+	}
+	rc.RespondWithApiError(err)
+}
+
+type boltPinger struct {
+	appEnv      *env.AppEnv
+	openReadTxs metrics.Gauge
+	running     concurrenz.AtomicBoolean
+}
+
+func (self *boltPinger) PingContext(ctx context.Context) error {
```

Comment:
Is anything using this? I can't find a reference to it. On first look, it seems like it doesn't belong at the REST level.

Reply:
It's in Register